### PR TITLE
Enable linker for AMP.navigateTo

### LINKED ISSUE #21170: amp linker won't decorate AMP.navigateTo()
AMP linker doesn't seems to decorate AMP.navigateTo destinations with CIDs.

If I have 
`<form id="myform" on="submit:AMP.navigateTo(url="MY/THANK_YOU/PAGE>")">`

The user won't be able to enter to that destination domain with a CID.

### PR DESCRIPTION
**Pull request by lannka**
Closes #21170

### CODE REVIEW

File: extensions/amp-analytics/0.1/linker-manager.js [hunk 1]
```diff
@@ -234,60 +240,61 @@ export class LinkerManager {
   }
 
   /**
-   * Called on click on any anchor element. Adds linker param if a match for
-   * given linker configuration.
-   * @param {!Element} element
-   * @param {!Event} event
+   * Apply linkers to the given url.
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/linker-manager.js]: Should we clarify that we might not append the linker params when this is called?

File: extensions/amp-analytics/0.1/test/test-linker-manager.js [hunk 2]
```diff
@@ -756,7 +773,7 @@ describes.realWin('Linker Manager', {amp: true}, env => {
       return linkerManager.init().then(() => {
         expect(beforeSubmitStub.calledOnce).to.be.true;
         toggleExperiment(win, 'linker-form', false);
-        return expect(beforeSubmitStub).calledWith(sinon.match.func);
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/test/test-linker-manager.js]: I thought these would silently fail without return? I could be wrong.

File: extensions/amp-analytics/0.1/test/test-linker-manager.js [hunk 3]
```diff
@@ -92,17 +99,17 @@ describes.realWin('Linker Manager', {amp: true}, env => {
       element
     ).init();
 
-    expect(handlers.length).to.equal(1);
+    expect(anchorClickHandlers).to.have.length(1);
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/test/test-linker-manager.js]: 👍I didn't know about this assertion.

File: extensions/amp-analytics/0.1/linker-manager.js [hunk 4]
```diff
@@ -117,8 +117,14 @@ export class LinkerManager {
...
-      navigation.registerAnchorMutator(
-        this.handleAnchorMutation_.bind(this),
+      navigation.registerAnchorMutator((element, event) => {
+        if (!element.href || event.type !== 'click') {
+          return;
+        }
+        element.href = this.applyLinkers_(element.href);
+      }, Priority.ANALYTICS_LINKER);
+      navigation.registerNavigateToMutator(
+        url => this.applyLinkers_(url),
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/linker-manager.js]: Is it guaranteed to get full url here, or is relative possible?

File: extensions/amp-analytics/0.1/linker-manager.js (hunk 1 above)
Comment (lannka) [reply extensions/amp-analytics/0.1/linker-manager.js]: done

File: extensions/amp-analytics/0.1/linker-manager.js (hunk 4 above)
Comment (lannka) [reply extensions/amp-analytics/0.1/linker-manager.js]: relative is possible. it does not affect our logic though.

File: extensions/amp-analytics/0.1/test/test-linker-manager.js (hunk 2 above)
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-linker-manager.js]: do you mean you had to use `return`? return is only necessary if the following statement is a promise.

Comment (calebcordry) [reply extensions/amp-analytics/0.1/test/test-linker-manager.js]: I just double-checked and you are right :)

**Pull request merged**